src/main/java/de/tum/in/www1/artemis/service/scheduled/DataExportScheduleService.java

```diff
@@ -68,9 +68,10 @@ public void createDataExportsAndDeleteOldOnes() {
 
         checkSecurityUtils();
         log.info("Creating data exports and deleting old ones");
-        Set<DataExport> successfulDataExports = new HashSet<>();
+        Set<DataExport> successfulDataExports = Collections.synchronizedSet(new HashSet<>());
         var dataExportsToBeCreated = dataExportRepository.findAllToBeCreated();
-        dataExportsToBeCreated.forEach(dataExport -> createDataExport(dataExport, successfulDataExports));
+        ExecutorService executor = Executors.newFixedThreadPool(10);
+        dataExportsToBeCreated.forEach(dataExport -> executor.execute(() -> createDataExport(dataExport, successfulDataExports)));
         var dataExportsToBeDeleted = dataExportRepository.findAllToBeDeleted();
         dataExportsToBeDeleted.forEach(this::deleteDataExport);
         Optional<User> admin = userService.findInternalAdminUser();
```

Comment:
Don't we need to wait here until the creation is finished before sending the failure-mail to the admin?

Reply:
The failure email is sent seperately for each data export. But I guess for the success email it's relevant because this is only sent once for all exports.